packages/web/components/bridge/immersive/crypto-fiat-input.tsx

```diff
@@ -46,6 +50,7 @@ export const CryptoFiatInput: FunctionComponent<{
   const inputRef = useRef<HTMLInputElement>(null);
 
   const [isMax, setIsMax] = useState(false);
+  const [hasSubtractedAmount, setHasSubtractedAmount] = useState(false);
```

Comment:
Wouldn't it be possible to derive this value instead? We could see if the input amount is greater then the max available amount and set `hasSubtractedAmount` to false if that's not the case. This could subject us to less invalid state / race conditions

Reply:
I considered this, but I'm not sure it's possible since we're changing the input value. Without a reference point, unless I keep a state value, it would be difficult to manage

Reply:
Resolved in my latest commit. It's much more streamlined, thank you!